src/Ng/CommandHelpers.cs

```diff
@@ -127,7 +127,10 @@ public static StorageFactory CreateSuffixedStorageFactory(string suffix, IDictio
             return CreateStorageFactoryImpl(arguments, names, verbose);
         }
 
-        private static StorageFactory CreateStorageFactoryImpl(IDictionary<string, string> arguments, IDictionary<string, string> argumentNameMap, bool verbose)
+        private static StorageFactory CreateStorageFactoryImpl(IDictionary<string, string> arguments, 
+                                                               IDictionary<string, string> argumentNameMap,
+                                                               bool verbose,
+                                                               bool compressed = false)
```

Comment:
All of the other arguments but `compressed` are passed to this function through a dictionary. By adding this additional argument as a parameter, you've created a separate source of arguments for the method. I would prefer the consistency of adding a `Arguments.StorageIsCompressed` to the dictionary, but ultimately this isn't a big deal and this implementation is fine.